### DISCUSSION TITLE
How to get HRRR temps AGL

### DISCUSSION THREAD

**Discussion by greggabott2** [blaylockbk/Herbie#309] [Q&A]
Hello. I am trying to calculate HRRR lapse rates (surface to 3km AGL). I cannot seem to figure out how to get 3km AGL temps to do this with.

**Comment by blaylockbk** [comment]
Thanks for your question. I haven't ever needed to do this before, so maybe someone else out there has a different idea how to do this.

NWP data is usually distributed on pressure levels. You could figure out the nearest pressure level corresponding to 3 km, and then get those pressure levels.

Using MetPy, 
https://unidata.github.io/MetPy/latest/api/generated/metpy.calc.height_to_pressure_std.html

```python
import metpy.calc as mcalc
from metpy.units import units

height = 3 * units.kilometers
mcalc.height_to_pressure_std(height)
```
3 km is approximatley 700 hPa in the standard atmosphere. So, you could get all the pressure levels from near the surface up to 700 hPa.

With Herbie, I can get the temperature at pressure levels larger than 700 hPa with this [regex](https://regex101.com/r/1Dku20/1) (I used ChatGPT to help me with the regex pattern "regex to get number >= 700 (including decimal number)").

```python
from herbie import Herbie

H = Herbie("2024-1-1", model="hrrr", product="prs")
H.inventory(r"TMP:(?:[7-9]\d{2}|[1-9]\d{3,})(?:\.\d+)? mb")
```
![image](https://github.com/blaylockbk/Herbie/assets/6249613/4032a6cc-9928-42f3-bada-434d34290247)
